lib/http.c

```diff
@@ -1456,8 +1456,10 @@ CURLcode Curl_http_done(struct connectdata *conn,
      data->state.negotiate.state == GSS_AUTHSENT) {
     /* add forbid re-use if http-code != 401/407 as a WA only needed for
      * 401/407 that signal auth failure (empty) otherwise state will be RECV
-     * with current code */
-    if((data->req.httpcode != 401) && (data->req.httpcode != 407))
+     * with current code.
+     * Do NOT close successful CONNECT-only connections. */
+    if((data->req.httpcode != 401) && (data->req.httpcode != 407) &&
+       (!data->set.connect_only || data->req.httpcode != 200))
```

Comment:
> Then in my opinion a corrected patch which omits the '|| data->req.httpcode
!= 200' part, will be good for time being (till we have a full fix for
negotiate).

Thanks, new patch here: #655